src/provider/provider_tracking.c

```diff
@@ -110,10 +110,10 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
     }
 
     ret = umfMemoryTrackerAdd(p->hTracker, p->pool, *ptr, size);
-    if (ret != UMF_RESULT_SUCCESS && p->hUpstream) {
-        if (umfMemoryProviderFree(p->hUpstream, *ptr, size)) {
-            // TODO: LOG
-        }
+    if (ret != UMF_RESULT_SUCCESS) {
```

Comment:
If allocation succeeds but `umfMemoryTrackerAdd` fails the `ret` will contain an error code even if the upstream memory provider returned `UMF_RESULT_SUCCESS`. The same applies to the free path below.

Reply:
Done. The free path below is OK.